### DISCUSSION TITLE
bind_map missing methods compared to native dicts

### DISCUSSION THREAD

**Discussion by kenoslund-google** [pybind/pybind11#3249] [Ideas]
When using bind_map from stl_bind.hpp, there are several standard (and widely used) methods that are missing compared to native dictionaries:

```
>>> print(set(dir({})) - set(dir(my_bound_map)))
{'values', 'popitem', 'get', 'pop', 'copy', 'fromkeys', 'update', 'keys', 'clear', 'setdefault'}
```

In particular, `values`, `keys` and `clear` are pretty widely used. Is there a reason these are missing? Could they be added? `keys` would probably be pretty trivial to implement with `make_key_iterator`. It doesn't look like an equivalent for `iter.second` exists, but presumably would be pretty simple to add following the same pattern as `make_key_iterator`. clear should be trivial too.

**Comment by Skylion007** [comment]
@kenoslund-google Just noone has bothered to implement them. PR welcome on this front with relevant tests.

**Reply by bmerry** [reply]
I ran into this too, so I've written https://github.com/pybind/pybind11/pull/3271 to add `make_value_iterator`, after which it should be very easy to add `keys` and `values`.

**Reply by bmerry** [reply]
Pybind 11 2.8.0 now has `keys` and `values`.
